frontend/src/components/Graph.tsx

```diff
@@ -44,36 +56,61 @@ const CustomGraph: React.FC = () => {
 
   const onMessage = (event: Event[]) => {
     const graph = graphRef.current
-
     const g6data: GraphData = dataParser(event)
     if (graph) {
-      g6data.nodes!.forEach((node) => {
+      g6data.nodes!.forEach((node: any) => {
+        const temp = node
+        const tempTime: number = temp.time
+        if (tempTime === time) {
+          temp.x = posx
+          if (posy < 0) {
+            temp.y = posy
+            posy = posy * -1 + 100 * 0.99 ** log
+            log += 1
+          } else {
+            temp.y = posy
+            if (posy !== 0) {
+              posy *= -1
+            }
+          }
+        } else if (tempTime > time) {
+          posx += 100
+          temp.x = posx
+          posy = 0
+          log = 1
+          temp.y = posy
+          posy += 100
+          time = tempTime
+        }
```

Comment:
can we put all of this inside a function and then call it? or it would work if you do it that way?

Reply:
I can try fixing it.